nodejs/aws-serverless/cloudwatch/eventRule.ts

```diff
@@ -0,0 +1,88 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as aws from "@pulumi/aws";
+import { cloudwatch, lambda } from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+import { createLambdaFunction, Handler } from "./../function";
+import { EventSubscription } from "./../subscription";
+
+/**
+ * Arguments to control the event rule subscription.  Currently empty, but still defined in case of
+ * future need.
+ */
+export interface EventRuleEventSubscriptionArgs {
+}
+
+export interface EventRuleEvent {
+    account: string;
+    region: string;
+    detail: any;
+    "detail-type": string;
+    source: string;
+    time: string;
+    id: string;
+    resources: string[];
+}
+
+export type EventRuleEventHandler = Handler<EventRuleEvent, void>;
+
+export function onSchedule(name: string, schedule: string, handler: EventRuleEventHandler, args?: EventRuleEventSubscriptionArgs, opts?: pulumi.ResourceOptions): EventRuleEventSubscription {
+    const eventRule = new cloudwatch.EventRule(name, { scheduleExpression: schedule }, opts);
+    return onEvent(name, eventRule, handler, args, opts);
+}
```

Comment:
@lukehoban This is an interesting entrypoint (that is different from basically everything else in the serverless project).  It doesn't operate on a resource per-se.  Instead, it is just passed a schedule string for convenience.  It's questionable to me how important this is, since someone can make the event-rule themselves and call the below method.  

Do you think we need this?

Reply:
From our discussion.  Not only do we want this, we likely want to break this down even further to onCronEvent and onRateEvent